src/sinks/splunk_hec/logs/tests.rs

```diff
@@ -389,3 +396,311 @@ fn splunk_encode_log_event_semantic_meanings() {
+    );
+
+    let sourcetype = Template::try_from("{{ event_sourcetype }}".to_string()).ok();
+    let source = Template::try_from("{{ event_source }}".to_string()).ok();
+    let index = Template::try_from("{{ event_index }}".to_string()).ok();
+    let indexed_fields = vec![
+        owned_value_path!("event_field1"),
+        owned_value_path!("event_field2"),
+    ];
+
+
+    let processed = process_log(
+        event,
+        &super::sink::HecLogData {
+            sourcetype: sourcetype.as_ref(),
+            source: source.as_ref(),
+            index: index.as_ref(),
+            host_key: Some(OptionalTargetPath {
+                path: Some(OwnedTargetPath::event(owned_value_path!("host_key"))),
+            }),
+            indexed_fields: indexed_fields.as_slice(),
+            endpoint_target: EndpointTarget::Event,
+            auto_extract_timestamp: false,
+            timestamp_configuration: None,
+        },
+    );
+
+    let hec_data =
+        get_encoded_event::<HecEventJson>(JsonSerializerConfig::default().into(), processed);
+
+    // Even though we didn't provide timestamp configuration, it takes user logspaced timestamp and
+    // adds that to the metadata, which will be the current time in seconds
+    assert_eq!(hec_data.time.is_some(), true);
+
+    // Timestamp key path is not present in event, we expect no ts_nanos_key
+    assert_eq!(hec_data.event.get("ts_nanos_key"), None);
+
+
+    // user_logspaced timestamp should be removed from the event
+    assert_eq!(
+        hec_data
+            .event
+            .get("timestamp")
+            .is_none(),
+        true
+    );
+
+    // Basic metadata checks
+    assert_eq!(hec_data.source, Some("test_source".to_string()));
+    assert_eq!(hec_data.sourcetype, Some("test_sourcetype".to_string()));
+    assert_eq!(hec_data.index, Some("test_index".to_string()));
+    assert_eq!(hec_data.host, Some("test_host".to_string()));
+    assert_eq!(hec_data.fields.get("event_field1").unwrap(), "test_value1");
+}
+
+fn create_test_event() -> Event {
+    let mut event = Event::Log(LogEvent::from("hello world"));
+    let log = event.as_mut_log();
+    log.insert("event_sourcetype", "test_sourcetype");
+    log.insert("event_source", "test_source");
+    log.insert("event_index", "test_index");
+    log.insert("host_key", "test_host");
+    log.insert("event_field1", "test_value1");
+    log.insert("event_field2", "test_value2");
+    log.insert("key", "value");
+    log.insert("int_val", 123);
+    event
+}
+
+#[test]
+fn test_timestamp_configurations() {
+    crate::test_util::trace_init();
+
+    struct TestCase {
+        name: &'static str,
+        timestamp_value: Value,
+        event_timestamp_insertion_path: String,
+        config: Option<TimestampConfiguration>,
+        expected_time: Option<f64>,
+        expected_nanos: Option<i64>,
+        time_metadata_should_exist: bool,
+        should_preserve_timestamp: bool,
+    }
+
+    let test_cases = vec![
```

Comment:
The idomatic way of doing this in Rust is to use `rstest`, it generates independent test names etc which makes it easy to see which case is failing.

Eg. (from chunked_gelf.rs)
```
    #[tokio::test]
    #[rstest]
    #[case::gzip(Compression::Gzip)]
    #[case::zlib(Compression::Zlib)]
    async fn decode_compressed_payload_with_no_decompression_decoder(
        #[case] compression: Compression,
    ) {
        let payload = "foo";
...
```